src/Compilers/CSharp/Portable/Symbols/Source/SourceExtensionTypeSymbol.cs

```diff
@@ -112,31 +112,27 @@ protected override void CheckBaseExtensions(BindingDiagnosticBag diagnostics)
                     }
                 }
             }
-        }
-
-        // PROTOTYPE consider caching and using a property like InterfacesAndTheirBaseInterfacesNoUseSiteDiagnostics
-        private MultiDictionary<NamedTypeSymbol, NamedTypeSymbol> GetExtensionsAndTheirBaseExtensionsNoUseSiteDiagnostics()
-        {
-            var baseExtensions = this.BaseExtensionsNoUseSiteDiagnostics;
-            var resultBuilder = new MultiDictionary<NamedTypeSymbol, NamedTypeSymbol>(baseExtensions.Length,
-                SymbolEqualityComparer.CLRSignature, SymbolEqualityComparer.ConsiderEverything);
 
-            foreach (var baseExtension in baseExtensions)
+            MultiDictionary<NamedTypeSymbol, NamedTypeSymbol> mapAllBaseExtensionsDuplicates()
```

Comment:
>mapAllBaseExtensionsDuplicates [](http://example.com/codeflow?start=62&length=30)

I am not sure what the name is supposed to convey. Could you please elaborate?

Reply:
The purpose of this helper is to build the map of duplicate extensions by CLR signature. I'm not sure how to best name it, but this seems better than the old name `GetExtensionsAndTheirBaseExtensionsNoUseSiteDiagnostics`...